### PR TITLE
FHAC-489:Commit AuditRepositoryCore EJB for asynchronous audit logging

### PR DESCRIPTION
**Pull request by alameluchidambaram**
AuditRepositoryCore EJB is incorporated in audit core to make asynchronous audit log call for request and responses.

### CODE REVIEW

File: Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/AuditLogger.java [hunk 1]
```diff
@@ -62,11 +64,13 @@
...
         String _interface, Boolean isRequesting, Properties webContextProperties, String serviceName) {
-
         LOG.trace("--- Before auditing of request message ---");
-        LogEventRequestType auditLogMsg = getAuditTransforms().transformRequestToAuditMsg(request, assertion, target,
-            direction, _interface, isRequesting, webContextProperties, serviceName);
-        auditLogMessages(auditLogMsg, assertion);
+        if (getAuditLogger() != null) {
+            getAuditLogger().auditRequestMessage(request, assertion, target, direction, _interface, isRequesting,
+                webContextProperties, serviceName, getAuditTransforms());
+        } else {
```
Comment (christophermay07) [inline Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/AuditLogger.java]: You already log an error if JNDI lookup fails, as part of getAuditLogger() -- do we need this else statement?

File: Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/ejb/impl/AuditEJBLoggerImpl.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+     * @param direction defines the Outbound/Inbound message
+     * @param _interface Entity, Adapter or Nwhin
+     * @param isRequesting true for initiator, false for responder
+     * @param webContextProperties Properties loaded from message context
+     * @param serviceName Name of the Service being audited
+     */
+    @Asynchronous
+    @Override
+    public void auditRequestMessage(T request, AssertionType assertion, NhinTargetSystemType target, String direction, String _interface, Boolean isRequesting, Properties webContextProperties, String serviceName, AuditTransforms transforms) {
+        LOG.trace("--- Before auditing of request message ---");
```
Comment (christophermay07) [inline Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/ejb/impl/AuditEJBLoggerImpl.java]: You use this exact log message in the AuditLogger class, do we need a statement both places?  If we do, can you make the message here (or in AuditLogger) more specific?

File: Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/ejb/impl/AuditEJBLoggerImpl.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+     * @param target target community
+     * @param direction defines the Outbound/Inbound message
+     * @param _interface Entity, Adapter or Nwhin
+     * @param isRequesting true for initiator, false for responder
+     * @param webContextProperties Properties loaded from message context
+     * @param serviceName Name of the Service being audited
+     */
+    @Asynchronous
+    @Override
+    public void auditRequestMessage(T request, AssertionType assertion, NhinTargetSystemType target, String direction, String _interface, Boolean isRequesting, Properties webContextProperties, String serviceName, AuditTransforms transforms) {
```
Comment (christophermay07) [inline Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/ejb/impl/AuditEJBLoggerImpl.java]: Please fix any code you've touched where the line length exceeds 120 characters.

File: Product/Production/Services/AuditRepositoryCore/src/main/resources/META-INF/ejb-jar.xml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ejb-jar xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xmlns="http://java.sun.com/xml/ns/javaee" xmlns:ejb="http://java.sun.com/xml/ns/javaee/ejb-jar_3_0.xsd"
```
Comment (christophermay07) [inline Product/Production/Services/AuditRepositoryCore/src/main/resources/META-INF/ejb-jar.xml]: ejb namespace is not needed here.

File: Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/AuditLogger.java (hunk 1 above)
Comment (jsmith2) [inline Product/Production/Services/AuditRepositoryCore/src/main/java/gov/hhs/fha/nhinc/audit/AuditLogger.java]: Agree, remove the else statement.

**Comment by jsmith2** [comment]
Reviewed, needs small changes and can be merged.

**Comment by christophermay07** [comment]
I've reviewed this PR, endorsing for merge without further comment.

**Pull request merged**